### PR TITLE
Split independent pricing fixes

### PR DESCRIPTION
**Pull request by alexmojaki**
## Summary
- fix pricing build compatibility imports
- add ModelPrice.is_free coverage and helper
- make update-prices success tests hermetic and fix snapshot cleanup ordering
- add independent JS regression tests for absent prices and mapped usage extraction

## Tests
- uv run pytest tests/test_price_calc.py tests/test_update_prices.py -q
- npm test --workspace=packages/js -- src/__tests__/calcPrice.test.ts src/__tests__/extractUsage.test.ts

---
## Summary by cubic
Stabilizes pricing builds and the background updater, and adds `ModelPrice.is_free()` to detect free models. Success-path tests now mock HTTP and cover concurrent stop behavior.

- **Bug Fixes**
  - Clear snapshot after the worker thread exits in `UpdatePrices.stop()` to avoid stale reinstalls during in-flight fetches.
  - Fix build/import compatibility: use `typing_extensions.TypeAlias` and `pydantic` `ValidationInfo`.
  - Remove an unused runtime free-price helper.

- **Tests**
  - Mock HTTP for success tests (no VCR), including default upstream URL; add wait-on-start and sync/async wait coverage.
  - Add an in-flight stop test to verify snapshot cleanup.
  - Add JS regression test for absent prices (treated as zero), plus Python tests for `ModelPrice.is_free()` on zero and tiered prices.

<sup>Written for commit 3fb7212323aee88291f68b91fbdd8627d19b39e9. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/js/src/__tests__/extractUsage.test.ts [hunk 1]
```diff
@@ -220,6 +220,35 @@ describe('extractUsage', () => {
...
+      const responseData = {
+        model: 'test-model',
+        usage: { cache_read_tokens: 200, input_tokens: 100 },
+      }
+
+      const { usage } = extractUsage(provider, responseData)
+
+      expect(usage).toEqual({ cache_read_tokens: 200, input_tokens: 100 })
+    })
+  })
```
Comment (alexmojaki) [inline packages/js/src/__tests__/extractUsage.test.ts]: are these 2 tests adding anything? do other tests not effectively cover this in both main and https://github.com/pydantic/genai-prices/pull/368?

File: packages/python/genai_prices/update_prices.py [hunk 2]
```diff
@@ -110,11 +110,11 @@ def stop(self):
         global _global_update_prices
 
         _global_update_prices = None
-        data_snapshot.set_custom_snapshot(None)
         if self._thread is not None:
             self._stop_event.set()
             self._thread.join()
             self._thread = None
+        data_snapshot.set_custom_snapshot(None)
```
Comment (alexmojaki) [inline packages/python/genai_prices/update_prices.py:118]: why?

File: tests/test_update_prices.py [hunk 3]
```diff
@@ -46,12 +60,10 @@ def fake_get(url: str, timeout: httpx.Timeout) -> Response:
...
 
 
-@pytest.mark.default_cassette('success.yaml')
-@pytest.mark.vcr()
-def test_update_prices_wait_on_start():
+def test_update_prices_wait_on_start(monkeypatch: pytest.MonkeyPatch):
+    _mock_update_prices_get(monkeypatch)
     assert data_snapshot._custom_snapshot is None
     with UpdatePrices() as update_prices:
-        assert data_snapshot._custom_snapshot is None
```
Comment (alexmojaki) [inline tests/test_update_prices.py:54]: why remove this

File: packages/python/genai_prices/update_prices.py (hunk 2 above)
Comment (alexmojaki) [reply packages/python/genai_prices/update_prices.py:118]: (AI) Addressed in 3fb7212

File: packages/js/src/__tests__/extractUsage.test.ts (hunk 1 above)
Comment (alexmojaki) [reply packages/js/src/__tests__/extractUsage.test.ts]: (AI) Addressed in 3fb7212

File: tests/test_update_prices.py (hunk 3 above)
Comment (alexmojaki) [reply tests/test_update_prices.py:54]: (AI) Removed because entering the context starts the background thread immediately. With the mocked fast fetch, the snapshot can be installed before the next line, so this assertion is timing-dependent. The test still checks the pre-context empty state and the post-wait populated state.

File: packages/js/src/__tests__/extractUsage.test.ts (hunk 1 above)
Comment (alexmojaki) [reply packages/js/src/__tests__/extractUsage.test.ts]: remove in the other pr

Comment (alexmojaki) [reply packages/js/src/__tests__/extractUsage.test.ts]: (AI) Addressed in 7d91d47 on PR #369.
